compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1318,13 +1318,44 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                         && let hir::IsAsync::NotAsync = self.tcx.asyncness(lifetime_ref.hir_id.owner.def_id)
                         && !self.tcx.features().anonymous_lifetime_in_impl_trait
                     {
-                        rustc_session::parse::feature_err(
-                            &self.tcx.sess.parse_sess,
-                            sym::anonymous_lifetime_in_impl_trait,
-                            lifetime_ref.span,
-                            "anonymous lifetimes in `impl Trait` are unstable",
-                        ).emit();
-                        return;
+
+                        match self.tcx.hir().get_generics(lifetime_ref.hir_id.owner.def_id) {
+                            Some(generics) => {
+                                for i in 0..generics.params.len()  {
+
+                                    if !generics.span.contains(generics.params[i].span)  {
+
+                                        let mut diag =  rustc_session::parse::feature_err(
+                                            &self.tcx.sess.parse_sess,
+                                            sym::anonymous_lifetime_in_impl_trait,
+                                            lifetime_ref.span,
+                                            "anonymous lifetimes in `impl Trait` are unstable",
+                                        );
+
+                                        diag.span_label(lifetime_ref.span, "expected named lifetime parameter");
+
+                                        diag.multipart_suggestion("consider introducing a named lifetime parameter",
+                                                                  vec![
+                                                                    (lifetime_ref.span.shrink_to_hi(), "'a ".to_owned()),
```

Comment:
This suggestion is only correct if you have a `&`, but it's a good start :smile:.  You can also have an elided lifetime with the `'_` syntax, or in a path `Ref<u8>` instead of `Ref<'_, u8>` for instance.